app/frontend/src/services/adminService.js

```diff
@@ -6,6 +6,18 @@ export default {
   // Form calls
   //
 
+  /**
+   * @function addFormUser
+   * Add a form user role (specific administrative task, for real form/user/role management see rbac service)
+   * @param {formId} formId The request body for the relationships
+   * @param {userId} userId The request body for the relationships
+   * @param {Object} [roles] The list of roles too add for this user
+   * @returns {Promise} An axios response
+   */
+  addFormUser(userId, formId, role) {
+    return appAxios().put(`${ApiRoutes.ADMIN}${ApiRoutes.FORMS}/${formId}/addUser`, { userId: userId, formId: formId, role: role });
+  },
```

Comment:
Function signature, jsdoc description and implementation are somewhat mismatched and misleading. There are two way to think of this service call:
1) We are only ever going to allow one user/form pairing happen for each call - can make it have multiple roles if we want. In this case, the implementation needs to be properly handling multiple roles...
```js
  addFormUser(userId, formId, roles) {
    return appAxios().put(`${ApiRoutes.ADMIN}${ApiRoutes.FORMS}/${formId}/addUser`, roles.map(role => ({
      userId: userId,
      formId: formId,
      role: role
    })));
  },
```
2) This service call can handle arbitrary user/form pairings, in addition to the multiple roles. In this case, addFormUser should just be `addFormUser(permissions)` where permissions is an array of those triplets and passes through wholesale instead.
I'd personally recommend for now going with option 1. Option 2 allows for more future flexibility, but the logic onus would be on the caller instead (likely the vuex method).

Reply:
I've modified it to support multiple roles as in 1